step-strikers-app/step-strikers-app/Classes/RPGCharacter.swift

```diff
@@ -0,0 +1,174 @@
+        self.weaponsInInventory = [fistsWeapon]
+        self.currWeapon = fistsWeapon
+        self.armorInInventory = [noArmorArmor]
+        self.currArmor = noArmorArmor
+    }
+    
+    // Universal Functions
+    func wield(weaponObject: Weapon){
+        if(weaponObject.checkIfProficient(wearer: self)){
+            self.currWeapon = weaponObject
+        } else {
+            print("\(self.characterName) is not proficient with \(weaponObject.name).\n They will use this weapon at a disadvantage")
+            self.currWeapon = weaponObject
+        }
+    }
+
+    func wear(armorObject: Armor){
+        if(armorObject.checkIfSuited(potentialWearer: self)){
+            self.currArmor = armorObject
+        } else {
+            print("\(armorObject.name) is not properly suited for \(self.characterName). Their armor class will be at a disadvantage")
+        }
+    }
+
+    func fight(target: inout RPGCharacter){
+        
+        let damageDealt = calculateDamage(wielder: self, target: target, damage: self.currWeapon.damage)
+        
+        decreaseHealth(amtDamage: damageDealt)
+
+        decreaseStamina()
+    }
+
+    func decreaseStamina(){
+        self.currStamina -= self.currWeapon.staminaCost
+        // TODO: Do something when exhausted
+    }
+
+    func decreaseHealth(amtDamage: Int){
+        self.currHealth -= amtDamage
+        // TODO: Do something when dead
+    }
+
+    func increaseStamina(amtIncrease: Int){
+        self.currStamina += amtIncrease
+        if(amtIncrease > self.maxStamina){
+            self.currStamina = self.maxStamina
+        }
+    }
+
+    func increaseHealth(amtIncrease: Int){
+        self.currHealth += amtIncrease
+        if(amtIncrease > self.maxHealth){
+            self.currHealth = self.maxHealth
+        }
+    }
+}
+
+
+class Caster: RPGCharacter {
+    var currSpellPoints: Int
+    var maxSpellPoints: Int
+    
+    // Rolling modifier
+    var spellModifier = 0
+    
+    init(characterName: String, userName: String, health: Int,
+         stamina: Int, spellPoints: Int) {
+        self.currSpellPoints = spellPoints
+        self.maxSpellPoints = spellPoints
+        super.init(characterName: characterName, userName: userName, health: health,
+                   stamina: stamina)
+    }
+    
+    func increaseSpellPoints(amtIncrease: Int){
+        self.currSpellPoints += amtIncrease
+        if(self.currSpellPoints > maxSpellPoints){
+            self.currSpellPoints = maxSpellPoints
+        }
+    }
+    
+    func decreaseSpellPoints(amtDecrease: Int){
+        self.currSpellPoints -= amtDecrease
+    }
+}
+    
+class Fighter: RPGCharacter {
+    override init(characterName: String, userName: String, health: Int,
+                  stamina: Int){
+        super.init(characterName: characterName, userName: userName, health: health, stamina: stamina)
+        
+        // Initialize Starter Inventory
+        self.weaponsInInventory = [fistsWeapon]
```

Comment:
Same as above, initialize with argument

Reply:
Re: above. All players start with no weapons nor armor

Reply:
Remember that every time the app exits, all of the data not in firebase is lost (so all of the objects created here are destroyed). So every time someone runs the app, the init methods will be called. So we read the current state of the character from firebase and pass it into these init functions. So we need to pass their saved inventory in or else they'll restart with no weapons and no armor every time they start the app.